test/encapsulated-child-logger-factory.test.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('..')
+
+test('encapsulates an child logger factory', async t => {
```

Comment:
Can we add one for testing when registering a plugin (not creating another child context but rather altering the parent)? When using `fastify-plugin`

Reply:
Yes, exactly what you suggested 👍

Reply:
Done